### ISSUE TITLE
HttpWebRequest generates unobserved exceptions for WebException

### ISSUE THREAD

**Issue by jahmai** [mono/mono#10729]
## Steps to Reproduce

1. Create a HttpWebRequest
2. Abort the request
3. GC.Collect() / GC.WaitForPendingFinalizers()
3. See that TaskScheduler.UnobservedTaskException is called

### Current Behavior

TaskScheduler.UnobservedTaskException is called with the aborted WebException

### Expected Behavior

TaskScheduler.UnobservedTaskException is not called.

## On which platforms did you notice this

Xamarin.iOS / Xamarin.Android

**Version Used**:

Xamarin.iOS 11.14
Xamarin.Android 9.0

## Stacktrace

170|01:26:12:134 (UTC)|D/CrashlyticsCore * Critical Unobserved task exception for 1. System.AggregateException: A Task's exception(s) were not observed either by Waiting on the Task or accessing its Exception property. As a result, the unobserved exception was rethrown by the finalizer thread. ---> System.Net.WebException: The request was aborted: The request was canceled. at System.Net.HttpWebRequest+<MyGetResponseAsync>d__243.MoveNext () [0x003d5] in <c5ed006c006c43fc9f89e54e9dfc50ca>:0 --- End of inner exception stack trace --- at System.Net.HttpWebRequest+<MyGetResponseAsync>d__243.MoveNext () [0x003d5] in <c5ed006c006c43fc9f89e54e9dfc50ca>:0 

I believe this is due to HttpWebRequest.responseTask being a WebCompletionSource that contains a TaskCompletionSource. It has it's exception property set, but is never 'observed'.

**Comment by marek-safar** [comment]
This is likely duplicate of #10488
